python/pyspark/sql/connect/functions.py

```diff
@@ -2373,6 +2374,117 @@ def hours(col: "ColumnOrName") -> Column:
 
 hours.__doc__ = pysparkfuncs.hours.__doc__
 
+
+def make_dt_interval(
+    days: Optional["ColumnOrName"] = None,
+    hours: Optional["ColumnOrName"] = None,
+    mins: Optional["ColumnOrName"] = None,
+    secs: Optional["ColumnOrName"] = None,
```

Comment:
it seems in this PR, users can use in this way
```
make_dt_interval(mins="col_a")  // skip `days`
```

while this usage is not supported on the Scala side.

Reply:
since `days`/`hours`/`minus` accepts `IntegerType` column, and `secs ` accepts `DecimalType`;
shall we make `days`/`hours`/`minus`: `Union["ColumnOrName", int]` and `secs` : `Union["ColumnOrName", float, decimal. Decimal]` ?

e.g. https://github.com/apache/spark/blob/747db6675da86e79d04e2fcc531b3b72b22ebf04/python/pyspark/sql/functions.py#L4757

Reply:
also cc @HyukjinKwon

Reply:
> since `days`/`hours`/`minus` accepts `IntegerType` column, and `secs ` accepts `DecimalType`; shall we make `days`/`hours`/`minus`: `Union["ColumnOrName", int]` and `secs` : `Union["ColumnOrName", float, decimal. Decimal]` ?
> 
> e.g.
> 
> https://github.com/apache/spark/blob/747db6675da86e79d04e2fcc531b3b72b22ebf04/python/pyspark/sql/functions.py#L4757

If we define it as `Union["ColumnOrName", int]` type, then we require users to write `days`, `hours`, `minus`, `secs` parameters simultaneously. Is this okay?

Because when users write
- make_dt_interval(days="col_a")

Will raise an exception.

Reply:
Or do we need to determine that when the `hours` parameter is not None, the previous days must not be None? Similar logic requires writing a lot of judgment logic.
Which one is better?

Reply:
WDYT? @HyukjinKwon

Reply:
I think we can keep current logic with :
`days/hours/minus: Union["ColumnOrName", int]`
`secs : Union["ColumnOrName", float, decimal. Decimal]`

Reply:
after second thought, l think we can start with `ColumnOrName`.

if we want to support other types, let's do it later

Reply:
Let me fix it.

Reply:
> it seems in this PR, users can use in this way
> 
> ```
> make_dt_interval(mins="col_a")  // skip `days`
> ```
> 
> while this usage is not supported on the Scala side.

@zhengruifeng 
In order to solve this problem, do we have to do the following?
<img width="713" alt="image" src="https://github.com/apache/spark/assets/15246973/3c62e095-8903-4609-836d-94f88a2a3cb1">

Reply:
The abnormal result is roughly like this:
<img width="873" alt="image" src="https://github.com/apache/spark/assets/15246973/89f00b8d-5d9b-4e07-891c-0c84516b0ee4">

Reply:
This is a relatively strict judgment, but it seems very heavy.

Reply:
yeah, let's keep current implementation.